src/main/java/org/cryptomator/common/vaults/Vault.java

```diff
@@ -178,13 +192,62 @@ public synchronized void lock(boolean forced) throws UnmountFailedException, IOE
 			mountHandle.mountObj().close();
 			mountHandle.specialCleanup().run();
 		} finally {
+			removeFromQuickAccess();
 			destroyCryptoFileSystem();
 		}
 
 		this.mountHandle.set(null);
 		LOG.info("Locked vault '{}'", getDisplayName());
 	}
 
+	private synchronized void addToQuickAccess() {
+		if (quickAccessEntry.get() != null) {
+			//we don't throw an exception since we don't wanna block unlocking
+			LOG.warn("Vault already added to quick access area. Will be removed on next lock operation.");
+			return;
+
+		}
+
+		quickAccessServices.stream() //
+				.filter(s -> s.getClass().getName().equals(settings.quickAccessService.getValue())) //
+				.findFirst() //
+				.ifPresentOrElse( //
+						this::addToQuickAccessInternal, //
+						() -> LOG.warn("Unable to add Vault to quick access area: Desired implementation not available.") //
+				);
+
+	}
+
+	private void addToQuickAccessInternal(@NotNull QuickAccessService s) {
+		if (getMountPoint() instanceof Mountpoint.WithPath mp) {
```

Comment:
should this be checked before choosing a quick access service?

Reply:
It does not matter. We can safely assume, that the list is not super long and takes a noticable amount of time to filter.

If we move the check to the calling method (`addToQuickAccess`), we would have to either add a paramter to the `add...Internal` function or duplicate code.